policy/release/github_certificate.rego

```diff
@@ -0,0 +1,121 @@
+	some extension in [_TRIGGER, _SHA, _NAME, _REPOSITORY, _REF]
+	not _fulcio_extension_value(extension)
+	result := lib.result_helper(rego.metadata.chain(), [extension.name])
+}
+
+# METADATA
+# title: GitHub Workflow Repository
+# description: >-
+#   Check if the value of the GitHub Workflow Repository extension in the image
+#   signature certificate matches one of the allowed values. Use the rule data
+#   key "allowed_gh_workflow_repos" to specify the list of allowed values.
+#   An empty allow list, which is the default value, causes this check to succeeded.
+# custom:
+#   short_name: gh_workflow_repository
+#   failure_msg: 'Repository %q not in allowed list: %v'
+#
+deny contains result if {
+	result := _check_extension(rego.metadata.chain(), "allowed_gh_workflow_repos", _REPOSITORY)
+}
+
+# METADATA
+# title: GitHub Workflow Repository
+# description: >-
+#   Check if the value of the GitHub Workflow Ref extension in the image
+#   signature certificate matches one of the allowed values. Use the rule data
+#   key "allowed_gh_workflow_refs" to specify the list of allowed values.
+#   An empty allow list, which is the default value, causes this check to succeeded.
+# custom:
+#   short_name: gh_workflow_ref
+#   failure_msg: 'Ref %q not in allowed list: %v'
+#
+deny contains result if {
+	result := _check_extension(rego.metadata.chain(), "allowed_gh_workflow_refs", _REF)
+}
+
+# METADATA
+# title: GitHub Workflow Name
+# description: >-
+#   Check if the value of the GitHub Workflow Name extension in the image
+#   signature certificate matches one of the allowed values. Use the rule data
+#   key "allowed_gh_workflow_names" to specify the list of allowed values.
+#   An empty allow list, which is the default value, causes this check to succeeded.
+# custom:
+#   short_name: gh_workflow_name
+#   failure_msg: 'Name %q not in allowed list: %v'
+#
+deny contains result if {
+	result := _check_extension(rego.metadata.chain(), "allowed_gh_workflow_names", _NAME)
+}
+
+# METADATA
+# title: GitHub Workflow Trigger
+# description: >-
+#   Check if the value of the GitHub Workflow Trigger extension in the image
+#   signature certificate matches one of the allowed values. Use the rule data
+#   key "allowed_gh_workflow_triggers" to specify the list of allowed values.
+#   An empty allow list, which is the default value, causes this check to succeeded.
+# custom:
+#   short_name: gh_workflow_trigger
+#   failure_msg: 'Trigger %q not in allowed list: %v'
+#
+deny contains result if {
+	result := _check_extension(rego.metadata.chain(), "allowed_gh_workflow_triggers", _TRIGGER)
+}
+
+_check_extension(chain, key, extension) := result if {
+	value := _fulcio_extension_value(extension)
+	allowed := lib.rule_data(key)
+	count(allowed) > 0
+	not value in allowed
+	result := lib.result_helper(chain, [value, allowed])
+}
+
+_certs contains cert if {
+	some sig in input.image.signatures
+	cert := crypto.x509.parse_certificates(sig.certificate)[0]
```

Comment:
I wonder if we can just rely on cosign doing this. Added them because it doesn't hurt (I think), but also added a TODO comment to verify this is already handled there.